toolchain/check/context.cpp

```diff
@@ -743,6 +735,16 @@ class TypeCompleter {
     llvm_unreachable("All builtin kinds were handled above");
   }
 
+  auto BuildImportRefUsedValueRepr(SemIR::TypeId type_id,
+                                   SemIR::ImportRefUsed import_ref) const
+      -> SemIR::ValueRepr {
+    const auto& import_ir = context_.import_irs().Get(import_ref.ir_id);
+    auto import_inst = import_ir->insts().Get(import_ref.inst_id);
+    CARBON_CHECK(import_inst.kind() != SemIR::InstKind::ImportRefUsed)
+        << "If ImportRefUsed can point at another, this would be recursive.";
```

Comment:
My guess is this will come up when re-exporting an import maybe? And then it might be valid to hit that here, but with a different `import_ir` to hop back through the chain? Or maybe that will have a different representation entirely?

(Clearly this isn't about this PR or blocking, just curious about the future here.)

Reply:
I'm thinking to have this happen the ImportRef may need to be in the export list directly. As a consequence, really it depends on the representation of the re-export. For example, if the re-export is via `alias`, it's likely the instruction we'd see exported is actually the `alias` rather than for the Imports. If you can re-export an entire library, we'd probably just do that by dumping more direct ImportRefs. So this logic might actually not be triggered.